include/nbl/builtin/hlsl/impl/tgmath_impl.hlsl

```diff
@@ -0,0 +1,767 @@
+#ifndef _NBL_BUILTIN_HLSL_TGMATH_IMPL_INCLUDED_
+#define _NBL_BUILTIN_HLSL_TGMATH_IMPL_INCLUDED_
+
+#include <nbl/builtin/hlsl/cpp_compat/basic.h>
+#include <nbl/builtin/hlsl/concepts.hlsl>
+#include <nbl/builtin/hlsl/concepts/vector.hlsl>
+#include <nbl/builtin/hlsl/spirv_intrinsics/core.hlsl>
+#include <nbl/builtin/hlsl/spirv_intrinsics/glsl.std.450.hlsl>
+#include <nbl/builtin/hlsl/ieee754.hlsl>
+
+namespace nbl
+{
+namespace hlsl
+{
+namespace tgmath_impl
+{
+
+template<typename T, typename U NBL_STRUCT_CONSTRAINABLE>
+struct lerp_helper;
+
+#ifdef __HLSL_VERSION
+#define MIX_FUNCTION spirv::fMix
+#else
+#define MIX_FUNCTION glm::mix
+#endif
```

Comment:
as per our discord conversation, every helper should have a FIRST partial specialization done with `is_same_v<decltype(spirv::...),expected_Return_type>` constraint in a `#ifdef __HLSL_VERSION` block

then the C++ versions and cross-language adaptors for emulated should come

Reply:
done here:
https://github.com/Devsh-Graphics-Programming/Nabla/pull/817